### PR TITLE
Decode events

### PR DESCRIPTION
**Pull request by gtandes**

### CODE REVIEW

File: app/api/account/[account]/mana/route.ts
Comment (roaminro) [inline app/api/account/[account]/mana/route.ts:1]: prettier issue

File: app/api/block/[block_id]/route.ts
Comment (roaminro) [inline app/api/block/[block_id]/route.ts:1]: prettier issue

File: app/api/decode/events/route.ts [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+import { revalidatePath } from 'next/cache'
+import { decodeEvents } from '@/utils/events'
+
+/**
+ * @swagger
+ * /api/decode/events:
+ *   post:
+ *     tags: [Decode]
+ *     description: This endpoint takes an array of "encoded" events and returns an array of "decoded" events.
+ *
```
Comment (roaminro) [inline app/api/decode/events/route.ts:13]: the parameter `- $ref: '#/components/parameters/X-JSON-RPC-URL'` is missing. It's a header parameter that's available on all routes. See this example on how to add it: https://github.com/roaminro/koinos-rest/blob/main/app/api/chain/id/route.ts#L11

File: app/api/decode/events/route.ts [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+ */
+
+export async function POST(request: NextRequest) {
+  try {
+    try {
+      const events = (await request.json()) as interfaces.EventData[]
+      const result = await decodeEvents(events)
+
+      const eventsPath = request.nextUrl.searchParams.get('events') || '/'
+      revalidatePath(eventsPath)
```
Comment (roaminro) [inline app/api/decode/events/route.ts:59]: remove cache revalidation related code

File: app/api/transaction/[transaction_id]/route.ts
Comment (roaminro) [inline app/api/transaction/[transaction_id]/route.ts:1]: prettier issue

File: app/swagger/page.tsx
Comment (roaminro) [inline app/swagger/page.tsx:1]: prettier issue

**Review by roaminro — COMMENTED** [review]
added review comments

File: app/api/transaction/[transaction_id]/route.ts
Comment (gtandes) [reply app/api/transaction/[transaction_id]/route.ts:1]: corrected

File: app/api/account/[account]/mana/route.ts
Comment (gtandes) [reply app/api/account/[account]/mana/route.ts:1]: corrected

File: app/api/block/[block_id]/route.ts
Comment (gtandes) [reply app/api/block/[block_id]/route.ts:1]: corrected

File: app/swagger/page.tsx
Comment (gtandes) [reply app/swagger/page.tsx:1]: corrected

File: app/api/decode/events/route.ts (hunk 2 above)
Comment (gtandes) [reply app/api/decode/events/route.ts:59]: removed

File: app/api/decode/events/route.ts (hunk 1 above)
Comment (gtandes) [reply app/api/decode/events/route.ts:13]: *     parameters:
 *      - $ref: '#/components/parameters/X-JSON-RPC-URL'
 
 Has been added to line 14 & 15

File: app/api/transaction/[transaction_id]/route.ts
Comment (roaminro) [inline app/api/transaction/[transaction_id]/route.ts:1]: Need to reconcile with version of this file that's on main

**Review by roaminro — COMMENTED** [review]
need to fix the transaction route, but it's looking good, almost there!

File: app/api/decode/events/route.ts (hunk 2 above)
Comment (roaminro) [reply app/api/decode/events/route.ts:62]: I think you forgot to the changes for that file, I can still see the revalidatePath on my end ;)

**Review by roaminro — COMMENTED** [review]
more comments

File: app/api/transaction/[transaction_id]/route.ts
Comment (roaminro) [reply app/api/transaction/[transaction_id]/route.ts:1]: It's missing a few bits, see file in main branch https://github.com/roaminro/koinos-rest/blob/main/app/api/transaction/%5Btransaction_id%5D/route.ts

File: app/api/decode/events/route.ts
Comment (roaminro) [inline app/api/decode/events/route.ts:1]: need to clean up imports

Comment (gtandes) [reply app/api/decode/events/route.ts:1]: by clean do you mean remove the unused imports?

**Review by roaminro — APPROVED** [review]

**Pull request closed without merging**